ir/shared/src/main/scala/org/scalajs/ir/Transformers.scala

```diff
@@ -277,4 +277,19 @@ object Transformers {
     }
   }
 
+  /** Transformer that only transforms in the local scope.
+   *
+   *  In practice, this means stopping at `Closure` boundaries: their
+   *  `captureValues` are transformed, but not their other members.
+   */
+  abstract class LocalScopeTransformer extends Transformer {
+    override def transform(tree: Tree): Tree = tree match {
+      case Closure(arrow, captureParams, params, restParam, body, captureValues) =>
+        Closure(arrow, captureParams, params, restParam, body,
+            transformTrees(captureValues))(tree.pos)
```

Comment:
Will it be faster with:
```scala
case closure: Closure => closure.copy(captureValues = transformTrees(captureValues))
```

Reply:
Who knows? Without benchmarking, we cannot know. That's not really the point, though. We always write our tree transformations using pattern extraction and explicit calls to `apply`. We almost never use `copy`, unless we have a good reason to.